src/controllers/summaryManagementController.js

```diff
@@ -0,0 +1,198 @@
+        .catch((errors) => { res.status(400).send(errors); });
+    })
+      .catch((error) => { res.status(400).send(error); });
+  };
+  const putSummaryGroup = function (req, res) {
+    /* if (!hasPermission(req.body.requestor.role, 'putTeam')) {
+      res.status(403).send('You are not authorized to make changes in the teams.');
+      return;
+    } */
+
+    const { summaryGroupId } = req.params;
+
+    summaryGroup.findById(summaryGroupId, (error, record) => {
+      if (error || record === null) {
+        res.status(400).send('No valid records found');
+        return;
+      }
+      record.summaryGroupName = req.body.summaryGroupName;
+      record.isActive = req.body.isActive;
+      record.createdDatetime = Date.now();
+      record.modifiedDatetime = Date.now();
+
+      record.save()
+        .then(results => res.status(201).send(results._id))
+        .catch(errors => res.status(400).send(errors));
+    });
+  };
+  const addTeamMemberToSummaryGroup = function (req, res) {
+    /* if (!hasPermission(req.body.requestor.role, 'putTeam')) {
+      res.status(403).send('You are not authorized to make changes in the teams.');
+      return;
+    } */
+
+    const { summaryGroupId } = req.params;
+
+    summaryGroup.findById(summaryGroupId, (error, record) => {
+      if (error || record === null) {
+        res.status(400).send('No valid records found');
+        return;
+      }
+
+      const teamMember = {
+        _id: req.body._id,
+        fullName: req.body.fullName,
+        role: req.body.role,
+      };
+
+      record.teamMembers.push(teamMember);
+      record.save()
+        .then(results => res.status(201).send(results._id))
+        .catch(errors => res.status(400).send(errors));
+    });
+  };
+
+
+  const getTeamMembersBySummaryGroupId = function (req, res) {
+    const { summaryGroupId } = req.params;
+
+    summaryGroup.findById(summaryGroupId, (error, summaryGroup) => {
+      if (error || !summaryGroup) {
+        return res.status(400).send('No valid records found');
+      }
+
+      const { teamMembers } = summaryGroup;
+      res.status(200).json({ teamMembers });
+    });
+  };
+  const deleteTeamMemberToSummaryGroup = function (req, res) {
+    const { summaryGroupId, userId } = req.params;
+
+
+    summaryGroup.findById(summaryGroupId, (error, record) => {
+      if (error || record === null) {
+        res.status(400).send({ error: 'No valid records found' });
+        return;
+      }
+      const teamMemberObjectId = mongoose.Types.ObjectId(userId);
+      const updatedTeamMembers = record.teamMembers.filter(teamMember => teamMember._id.toString() !== teamMemberObjectId.toString());
+
+      record.teamMembers = updatedTeamMembers;
+      record.save()
+      .then(results => res.status(204).send())
+      .catch(errors => res.status(400).send(errors));
+    });
+  };
+  const addSummaryReceiversToSummaryGroup = function (req, res) {
```

Comment:
Similar to what nathanah suggested, you should check to see if user is already a summary receiver to prevent duplicate users from being added.

Reply:
Hi @lchua2314 ,
I have already change the duplicate user problem in front end.Thanks